### PR TITLE
(kubernetes management) Updatecli securitygroups from AWS 

### PR DESCRIPTION
**Pull request by smerle33**
Checking the version the Security Group from AWS used by kubernetes.

Still WIP  @dduportal

### CODE REVIEW

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  getNewJenkinsInfraSecurityGroup:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/jenkins-infra/aws/main/ec2-agents.tf
+      #matchPattern: 'aws_security_group.*(?:\n|\n\r).*name.*"(.*)".*(?:\n|\n\r).*infra\.ci'
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:20]:
```suggestion
```

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      #- trimPrefix: "name.*="
+  getNewJenkinsReleaseSecurityGroup:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/jenkins-infra/aws/main/ec2-agents.tf
+      #matchPattern: 'aws_security_group.*(?:\n|\n\r).*name.*"(.*)".*(?:\n|\n\r).*release\.ci'
+      matchPattern: "aws_security_group.*(\n|\n\r).*name.*\"(.*)\".*(\n|\n\r).*release\\.ci"
+    transformers:
+      - findSubMatch: 'name.*"(.*)"'
+      #- trimPrefix: "name.*="
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:33]:
```suggestion
      matchPattern: "aws_security_group.*(\n|\n\r).*name.*\"(.*)\".*(\n|\n\r).*release\\.ci"
    transformers:
      - findSubMatch: 'name.*"(.*)"'
```

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    scmID: default
+  setJenkinsReleaseSecurityGroup:
+    name: "Bump Release Security Group"
+    kind: file
+    sourceID: getNewJenkinsReleaseSecurityGroup
+    spec:
+      file: config/ext_jenkins-release.yaml
+      matchPattern: 'securityGroups: ".*"'
+      replacePattern: 'securityGroups: "{{ source `getNewJenkinsReleaseSecurityGroup` }}"'
+      #TODO check if all the occurence are changed (3) or just the first one
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:54]:
```suggestion
    scmID: default
  setJenkinsReleaseSecurityGroup:
    name: "Bump Release Security Group"
    kind: file
    sourceID: getNewJenkinsReleaseSecurityGroup
    spec:
      file: config/ext_jenkins-release.yaml
      matchPattern: 'securityGroups: ".*"'
      replacePattern: 'securityGroups: "{{ source `getNewJenkinsReleaseSecurityGroup` }}"'
```
It's a "replace all"

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    scmID: default
+
+pullrequests:
+  default:
+    kind: github
+    scmID: default
+    targets:
+      - setJenkinsInfraSecurityGroup
+      - setJenkinsReleaseSecurityGroup
+
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:64]:
```suggestion
      - setJenkinsReleaseSecurityGroup

```

**Comment by lemeurherve** [comment]
Note: you can add all these suggestions in "batch suggestion" when you're on the "files" tab in order to create only one commit with all of them instead of one commit for each.

File: updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          values: "jenkins-agent-windows-2019-amd64-*"
+        - name: "tag:build_type"
+          values: "prod"
+        - name: "tag:version"
+          values: '{{ source "packerImageVersion" }}'
+  getLatestUbuntuAgentAMIArm64:
+    kind: aws/ami
+    depends_on:
+      - packerImageVersion
+    sortBy: creationDateAsc
```
Comment (olblak) [inline updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml:54]: sortBy should be one level below in the spec

File: updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml [hunk 6]
```diff
@@ -99,3 +99,4 @@ pullrequests:
       - setUbuntuAgentAMIAmd64
       - setUbuntuAgentAMIArm64
       - setWindowsAgentAMIamd64
+
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml:102]:
```suggestion
```

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      replacePattern: 'securityGroups: "{{ source `getNewJenkinsReleaseSecurityGroup` }}"'
+    scmID: default
+
+pullrequests:
+  default:
+    kind: github
+    scmID: default
+    targets:
+      - setJenkinsInfraSecurityGroup
+      - setJenkinsReleaseSecurityGroup
```
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:63]:
```suggestion
      - setJenkinsReleaseSecurityGroup
```

**Review by lemeurherve — APPROVED** [review]

File: updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml (hunk 7 above)
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-infra-and-release-securitygroups.yaml:63]:
```suggestion
      - setJenkinsReleaseSecurityGroup

```

File: updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml (hunk 6 above)
Comment (lemeurherve) [inline updatecli/updatecli.d/charts/jenkins-agents-infra-release.yaml:102]:
```suggestion
```

**Review by lemeurherve — APPROVED** [review]

**Review by dduportal — APPROVED** [review]
```
[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1509)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1510)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1511)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1512)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1513)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1514)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1515)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1516)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1517)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1518)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1519)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1520)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1521)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1522)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1523)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1524)PULL REQUESTS

=============




#####################################################

# BUMP SECURITY GROUPS ON JENKINS-INFRA AND RELEASE #

#####################################################



SOURCES

=======

Validating "JENKINS-INFRA-AND-RELEASE-SECURITYGROUPS.YAML"


getNewJenkinsReleaseSecurityGroup

---------------------------------

✔ Content: found from file "https://raw.githubusercontent.com/jenkins-infra/aws/main/locals.tf[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1525)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1526)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1527)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1528)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1529)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1530)":

aws_security_groups = ["infraci:infra.ci.jenkins.io:20.72.105.159/32", "release:

Validating "JENKINS-INFRA-AND-RELEASE-SECURITYGROUPS.YAML"


getNewJenkinsInfraSecurityGroup

-------------------------------

✔ Content: found from file "https://raw.githubusercontent.com/jenkins-infra/aws/main/locals.tf[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1531)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1532)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1533)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1534)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1535)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1536)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1537)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1538)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1539)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1540)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1541)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1542)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1543)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1544)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1545)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1546)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1547)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1548)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1549)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1550)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1551)[](https://infra.ci.jenkins.io/blue/organizations/jenkins/kubernetes-management/detail/PR-1983/10/pipeline/60#step-64-log-1552)":

aws_security_groups = ["infraci:



TARGETS

========

Validating "JENKINS-INFRA-AND-RELEASE-SECURITYGROUPS.YAML"


setJenkinsInfraSecurityGroup

----------------------------


**Dry Run enabled**


✔ Content from file "/tmp/updatecli/jenkins-infra/kubernetes-management/config/ext_jenkins-infra.yaml" already up to date

Validating "JENKINS-INFRA-AND-RELEASE-SECURITYGROUPS.YAML"


setJenkinsReleaseSecurityGroup

------------------------------


**Dry Run enabled**


✔ Content from file "/tmp/updatecli/jenkins-infra/kubernetes-management/config/ext_jenkins-release.yaml" already up to date
```


👏 looks great !

**Pull request merged**